File: src/main/java/es/qky/selenidepom/RequiredError.java
package es.qky.selenidepom;

import com.codeborne.selenide.WebDriverRunner;
import com.codeborne.selenide.ex.UIAssertionError;

import java.util.ArrayList;
import java.util.List;


/**
 * Error thrown when a @Required element is not found.
 */
public class RequiredError extends UIAssertionError {
    final List<Throwable> errors;

    /**
     * Constructor.
     * @param errors List of Throwable errors found while looking for @Required elements.
     */
    public RequiredError(List<Throwable> errors) {
        super(WebDriverRunner.driver(), "Required elements not found in page: " + errors.size() + " errors. See suppressed errors");
        this.errors = errors;
        errors.forEach(this::addSuppressed);
    }

    /**
     * Accessor.
     * @return errors list.
     */
    public List<Throwable> getErrors() {
        return new ArrayList<>(errors);
    }
}

File: src/main/kotlin/com/github/qky666/selenidepom/error/RequiredError.kt
package com.github.qky666.selenidepom.error

import com.codeborne.selenide.ex.UIAssertionError

/**
 * Error thrown when a [com.github.qky666.selenidepom.annotation.Required] element is not found.
 */
class RequiredError(errors: List<Throwable>) :
    UIAssertionError("Required elements not found in page: ${errors.size} errors. See suppressed errors") {
    init {
        errors.forEach { this.addSuppressed(it) }
    }
}
